### DISCUSSION TITLE
how to create a RGB composite using multiband reader

### DISCUSSION THREAD

**Discussion by iferencik** [developmentseed/titiler#497] [General]
I have a custom Multiband reader which I am feeding 3 urls representing 3 images

The multiple URL's are B4, B3 and B2 bands of a sentinel 2 image

here is the [info](https://titiler.undpgeohub.org/cogs/info?url=https%3A%2F%2Fsentinel2l2a01.blob.core.windows.net%2Fsentinel2-l2%2F47%2FX%2FML%2F2022%2F07%2F15%2FS2A_MSIL2A_20220715T074621_N0400_R135_T47XML_20220715T141448.SAFE%2FGRANULE%2FL2A_T47XML_A036883_20220715T074624%2FIMG_DATA%2FR10m%2FT47XML_20220715T074621_B04_10m.tif%3Fc3Q9MjAyMi0wNy0xNFQxNiUzQTM2JTNBNDFaJnNlPTIwMjItMDctMTVUMTclM0EyMSUzQTQxWiZzcD1ybCZzdj0yMDIxLTA2LTA4JnNyPWMmc2tvaWQ9Yzg1YzE1ZDYtZDFhZS00MmQ0LWFmNjAtZTJjYTBmODEzNTliJnNrdGlkPTcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyZza3Q9MjAyMi0wNy0xNVQwNyUzQTM5JTNBMTlaJnNrZT0yMDIyLTA3LTIyVDA3JTNBMzklM0ExOVomc2tzPWImc2t2PTIwMjEtMDYtMDgmc2lnPUdMTmxiRUV1biUyQlJGN3l4WWF5bGFBVFdsSnZVcU1CTEZaZ2FQbFVOUE9MOCUzRA%3D%3D&url=https%3A%2F%2Fsentinel2l2a01.blob.core.windows.net%2Fsentinel2-l2%2F47%2FX%2FML%2F2022%2F07%2F15%2FS2A_MSIL2A_20220715T074621_N0400_R135_T47XML_20220715T141448.SAFE%2FGRANULE%2FL2A_T47XML_A036883_20220715T074624%2FIMG_DATA%2FR10m%2FT47XML_20220715T074621_B03_10m.tif%3Fc3Q9MjAyMi0wNy0xNFQxNiUzQTM2JTNBNDFaJnNlPTIwMjItMDctMTVUMTclM0EyMSUzQTQxWiZzcD1ybCZzdj0yMDIxLTA2LTA4JnNyPWMmc2tvaWQ9Yzg1YzE1ZDYtZDFhZS00MmQ0LWFmNjAtZTJjYTBmODEzNTliJnNrdGlkPTcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyZza3Q9MjAyMi0wNy0xNVQwNyUzQTM5JTNBMTlaJnNrZT0yMDIyLTA3LTIyVDA3JTNBMzklM0ExOVomc2tzPWImc2t2PTIwMjEtMDYtMDgmc2lnPUdMTmxiRUV1biUyQlJGN3l4WWF5bGFBVFdsSnZVcU1CTEZaZ2FQbFVOUE9MOCUzRA%3D%3D&url=https%3A%2F%2Fsentinel2l2a01.blob.core.windows.net%2Fsentinel2-l2%2F47%2FX%2FML%2F2022%2F07%2F15%2FS2A_MSIL2A_20220715T074621_N0400_R135_T47XML_20220715T141448.SAFE%2FGRANULE%2FL2A_T47XML_A036883_20220715T074624%2FIMG_DATA%2FR10m%2FT47XML_20220715T074621_B02_10m.tif%3Fc3Q9MjAyMi0wNy0xNFQxNiUzQTM2JTNBNDFaJnNlPTIwMjItMDctMTVUMTclM0EyMSUzQTQxWiZzcD1ybCZzdj0yMDIxLTA2LTA4JnNyPWMmc2tvaWQ9Yzg1YzE1ZDYtZDFhZS00MmQ0LWFmNjAtZTJjYTBmODEzNTliJnNrdGlkPTcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyZza3Q9MjAyMi0wNy0xNVQwNyUzQTM5JTNBMTlaJnNrZT0yMDIyLTA3LTIyVDA3JTNBMzklM0ExOVomc2tzPWImc2t2PTIwMjEtMDYtMDgmc2lnPUdMTmxiRUV1biUyQlJGN3l4WWF5bGFBVFdsSnZVcU1CTEZaZ2FQbFVOUE9MOCUzRA%3D%3D&bands=b1&bands=b2&bands=b3) endpoint

folowing [url](https://titiler.undpgeohub.org/cogs/tiles/7/97/12?url=https%3A%2F%2Fsentinel2l2a01.blob.core.windows.net%2Fsentinel2-l2%2F47%2FX%2FML%2F2022%2F07%2F15%2FS2A_MSIL2A_20220715T074621_N0400_R135_T47XML_20220715T141448.SAFE%2FGRANULE%2FL2A_T47XML_A036883_20220715T074624%2FIMG_DATA%2FR10m%2FT47XML_20220715T074621_B04_10m.tif%3Fc3Q9MjAyMi0wNy0xNFQxNiUzQTM2JTNBNDFaJnNlPTIwMjItMDctMTVUMTclM0EyMSUzQTQxWiZzcD1ybCZzdj0yMDIxLTA2LTA4JnNyPWMmc2tvaWQ9Yzg1YzE1ZDYtZDFhZS00MmQ0LWFmNjAtZTJjYTBmODEzNTliJnNrdGlkPTcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyZza3Q9MjAyMi0wNy0xNVQwNyUzQTM5JTNBMTlaJnNrZT0yMDIyLTA3LTIyVDA3JTNBMzklM0ExOVomc2tzPWImc2t2PTIwMjEtMDYtMDgmc2lnPUdMTmxiRUV1biUyQlJGN3l4WWF5bGFBVFdsSnZVcU1CTEZaZ2FQbFVOUE9MOCUzRA%3D%3D&url=https%3A%2F%2Fsentinel2l2a01.blob.core.windows.net%2Fsentinel2-l2%2F47%2FX%2FML%2F2022%2F07%2F15%2FS2A_MSIL2A_20220715T074621_N0400_R135_T47XML_20220715T141448.SAFE%2FGRANULE%2FL2A_T47XML_A036883_20220715T074624%2FIMG_DATA%2FR10m%2FT47XML_20220715T074621_B03_10m.tif%3Fc3Q9MjAyMi0wNy0xNFQxNiUzQTM2JTNBNDFaJnNlPTIwMjItMDctMTVUMTclM0EyMSUzQTQxWiZzcD1ybCZzdj0yMDIxLTA2LTA4JnNyPWMmc2tvaWQ9Yzg1YzE1ZDYtZDFhZS00MmQ0LWFmNjAtZTJjYTBmODEzNTliJnNrdGlkPTcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyZza3Q9MjAyMi0wNy0xNVQwNyUzQTM5JTNBMTlaJnNrZT0yMDIyLTA3LTIyVDA3JTNBMzklM0ExOVomc2tzPWImc2t2PTIwMjEtMDYtMDgmc2lnPUdMTmxiRUV1biUyQlJGN3l4WWF5bGFBVFdsSnZVcU1CTEZaZ2FQbFVOUE9MOCUzRA%3D%3D&url=https%3A%2F%2Fsentinel2l2a01.blob.core.windows.net%2Fsentinel2-l2%2F47%2FX%2FML%2F2022%2F07%2F15%2FS2A_MSIL2A_20220715T074621_N0400_R135_T47XML_20220715T141448.SAFE%2FGRANULE%2FL2A_T47XML_A036883_20220715T074624%2FIMG_DATA%2FR10m%2FT47XML_20220715T074621_B02_10m.tif%3Fc3Q9MjAyMi0wNy0xNFQxNiUzQTM2JTNBNDFaJnNlPTIwMjItMDctMTVUMTclM0EyMSUzQTQxWiZzcD1ybCZzdj0yMDIxLTA2LTA4JnNyPWMmc2tvaWQ9Yzg1YzE1ZDYtZDFhZS00MmQ0LWFmNjAtZTJjYTBmODEzNTliJnNrdGlkPTcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyZza3Q9MjAyMi0wNy0xNVQwNyUzQTM5JTNBMTlaJnNrZT0yMDIyLTA3LTIyVDA3JTNBMzklM0ExOVomc2tzPWImc2t2PTIwMjEtMDYtMDgmc2lnPUdMTmxiRUV1biUyQlJGN3l4WWF5bGFBVFdsSnZVcU1CTEZaZ2FQbFVOUE9MOCUzRA%3D%3D&bands=b1&bands=b2&bands=b3) produces a gray image

![12](https://user-images.githubusercontent.com/92793531/179288047-c9502062-23a8-42a0-88a2-d1df0b7c1333.png)

I'd like to produce something like [this](https://planetarycomputer.microsoft.com/api/data/v1/item/tiles/WebMercatorQuad/10/867/560@1x?color_formula=Gamma+RGB+3.7+Saturation+1.5+Sigmoidal+RGB+15+0.35&nodata=0&collection=sentinel-2-l2a&item=S2A_MSIL2A_20211216T015621_R117_T51KXB_20211216T153808&assets=B04&assets=B03&assets=B02)

![560@1x](https://user-images.githubusercontent.com/92793531/179288272-8c3b9867-4548-4a8a-9872-57cdd6f74895.png)

Is is possible to produce a composite using the multiband reader?

the FastAPI docs of the router are here

https://titiler.undpgeohub.org/docs#/Multibands%20Cloud%20Optimized%20GeoTIFF

Thanks

**Comment by vincentsarago** [comment]
@iferencik you need to apply either rescaling or color formula. The sentinel-2 data is in uint16 so it needs to be rescaled to 0->255 in order to create a JPEG or PNG image. By default TiTiler will rescale the data using the min/max of the **data type**
